### PR TITLE
feat: functional build for darwin

### PR DESCRIPTION
**Pull request by kevinwright**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,6 +79,21 @@ To launch the interactive CLI, run the executable without any arguments.
 
 Run `ficsit help` to see a list of available commands.
 
+## Game Installations
+
+The app needs to know where the game is installed. On Steam, this will be
+
+    C:\Program Files (x86)\Steam\steamapps\common\Satisfactory
```
Comment (Th3Fanbus) [inline README.md:86]: Not always. Steam installations should be auto-detected anyway

File: README.md [hunk 2]
```diff
@@ -78,6 +79,21 @@ To launch the interactive CLI, run the executable without any arguments.
 
 Run `ficsit help` to see a list of available commands.
 
+## Game Installations
+
+The app needs to know where the game is installed. On Steam, this will be
```
Comment (budak7273) [inline README.md:84]: I thought it could auto-detect some common installs? Or maybe I'm thinking of it picking up on installs that SMM found in the past

File: README.md (hunk 1 above)
Comment (kevinwright) [reply README.md:86]: Not on MacOS they aren't

File: README.md [hunk 3]
```diff
@@ -78,6 +79,21 @@ To launch the interactive CLI, run the executable without any arguments.
...
+
+    C:\Program Files (x86)\Steam\steamapps\common\Satisfactory
+
```
Comment (budak7273) [inline README.md:87]: - This section is missing mention of Linux
- That's the default steam path, but it depends on where the user installed it
- I would rather link to https://docs.ficsit.app/satisfactory-modding/latest/faq.html#Files_GameInstall than list them all here. If we do that, we need to add the mac location to the FAQ page.

File: README.md [hunk 4]
```diff
@@ -78,6 +79,21 @@ To launch the interactive CLI, run the executable without any arguments.
...
+For MacOS and Crossover use (Assuming your bottle is named "Steam"):
+
+    ~/Library/Application\ Support/CrossOver/Bottles/Steam/drive_c/Program\ Files\ \(x86\)/Steam/steamapps/common/Satisfactory
+
+The command to add an installation is then:
```
Comment (budak7273) [inline README.md:92]:
```suggestion
To add an installation, either use the interactive interface, or use the command:
```

**Review by budak7273 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (mircearoata) [reply README.md:86]: > Not always. Steam installations should be auto-detected anyway

This is ficsit-cli, not SMM. Only SMM auto-detects installs

File: README.md (hunk 2 above)
Comment (budak7273) [reply README.md:84]:
```suggestion
The app needs to know where the game is installed.
On Steam, this depends on your chosen location, but by default is 
```

File: README.md (hunk 3 above)
Comment (kevinwright) [reply README.md:86]: - I don't know what the path would be on Linux
- Nothing in this README even mentions that FAQ page, how would anyone even know to look there?
- Forcing a person to dig through an entire FAQ for the one piece of information they **need** to make this app work is overkill
- Fair point, I'll change the text to clarify this is only the default path

Comment (budak7273) [reply README.md:86]: "How would anyone even know to look there" because this readme would be changed to link them to it

"Forcing a person to dig through an entire FAQ" please note that the link I posted takes viewers directly to the relevant heading

Comment (budak7273) [reply README.md:86]: Added Mac path to FAQ in https://github.com/satisfactorymodding/Documentation/commit/8f7e73a1a25c197f0d804fd3633a0e388cfa25f8

Comment (kevinwright) [reply README.md:86]: That FAQ doesn't tell me the default. Nor does it give me enough of a hint to work out what my own path would be if I had installed to a custom location without forcing me to go through the Steam UI.

Remember... we're talking here about folk with the technical literacy to both install to a custom location in the first place and to favour a command-line tool over SMM. For this audience in particular "here is the default path" is the only information they need.

File: README.md [hunk 5]
```diff
@@ -78,6 +78,25 @@ To launch the interactive CLI, run the executable without any arguments.
...
+
+When running on Windows, it should auto-detect this path and you have nothing to worry about.   
+
+But this won't be the case When running on MacOS with Satisfactory installed in Crossover.
+In this case you have to add the installation manually.
+The path to a crossover-installed game, assuming a bottle named "Steam" will be
+
+    ~/Library/Application\ Support/CrossOver/Bottles/Steam/drive_c/Program\ Files\ \(x86\)/Steam/steamapps/common/Satisfactory
+
+The command to add an installation is then:
```
Comment (budak7273) [inline README.md:95]:
```suggestion
The app needs to know where the game is installed to function. 

If you're not sure where that is, see the [Satisfactory Modding FAQ section about locating your game install](https://docs.ficsit.app/satisfactory-modding/latest/faq.html#Files_GameInstall) to learn approaches for finding out.

To add an installation, either use the interactive interface, or use the command:
```

File: README.md (hunk 4 above)
Comment (budak7273) [reply README.md:95]: Superseded by https://github.com/satisfactorymodding/ficsit-cli/pull/61/files#r1562711449

File: README.md (hunk 3 above)
Comment (budak7273) [reply README.md:87]: See https://github.com/satisfactorymodding/ficsit-cli/pull/61/files#r1562711449

File: README.md (hunk 2 above)
Comment (budak7273) [reply README.md]: Superseded by https://github.com/satisfactorymodding/ficsit-cli/pull/61/files#r1562711449

File: README.md (hunk 3 above)
Comment (kevinwright) [reply README.md:87]: That appears to just be a link back to this PR

Comment (budak7273) [reply README.md:87]: What hint do you suggest using providing to work out a custom install location?

Comment (budak7273) [reply README.md:87]: > That appears to just be a link back to this PR

My bad, not sure why that link broke, here's a new one: https://github.com/satisfactorymodding/ficsit-cli/pull/61#discussion_r1562711449

File: README.md (hunk 5 above)
Comment (kevinwright) [reply README.md]: That FAQ **DOES NOT HELP** on MacOS.  It still only shows the path in Windows-format as it appears *inside the bottle*, it still forces people to manually translate that path to the path on the MacOS filesystem containing that bottle, and it doesn't give a quick-and-easy example of the default path that 99% of users will see.

If you want to reject this PR on that basis then so be it, but I'm not prepared to accept a proposed change that **WILL** make this tool harder to use on MacOS.

File: README.md (hunk 3 above)
Comment (kevinwright) [reply README.md:87]: > What hint do you suggest using providing to work out a custom install location?

The default path **IS** the hint.

It tells me everything I need to know.

- It tells me I want a path pointing to the `Satisfactory` folder (and not, say, the `FactoryGame` folder underneath that). This is true even in a custom steam folder
- It tells me to look under `${HOME}/Library/Application Support/CrossOver/Bottles/<bottle name>/drive_c` on MacOS

For someone used to working with command line tools and comfortable doing so, that's all you really need.

File: README.md (hunk 4 above)
Comment (budak7273) [reply README.md:95]: > The interactive interface doesn't actually have a way to add an installation.

It does, through the manage installs screen, unless it has changed since I last worked on this project

File: README.md (hunk 3 above)
Comment (kevinwright) [reply README.md:87]: This PR, as it stands, is entirely self-contained with everything needed to build and run the app and install mods, without any need for external reference to an FAQ that is clearly written to hand-hold users with a lower degree of technical competence.

That's not a property I'm willing to sacrifice.

File: cmd/root.go [hunk 6]
```diff
@@ -131,6 +131,8 @@ func init() {
 		baseLocalDir = os.Getenv("APPDATA")
 	case "linux":
 		baseLocalDir = filepath.Join(os.Getenv("HOME"), ".local", "share")
+	case "darwin":
+		baseLocalDir = filepath.Join(os.Getenv("HOME"), ".ficsit-cli")
```
Comment (mircearoata) [inline cmd/root.go:135]: The standard for this seems to be `~/Library/Application Support`, so we should use that instead. Also `baseLocalDir` is the path without the app name, that's added below.

File: tools.go [hunk 7]
```diff
@@ -46,6 +46,8 @@ func main() {
 		baseLocalDir = os.Getenv("APPDATA")
 	case "linux":
 		baseLocalDir = filepath.Join(os.Getenv("HOME"), ".local", "share")
+	case "darwin":
+		baseLocalDir = filepath.Join(os.Getenv("HOME"), ".ficsit-cli")
```
Comment (mircearoata) [inline tools.go:50]: Same thing as for the other `baseLocalDir`

File: cmd/root.go (hunk 6 above)
Comment (kevinwright) [reply cmd/root.go]: done

File: tools.go (hunk 7 above)
Comment (kevinwright) [reply tools.go]: done

File: README.md (hunk 3 above)
Comment (budak7273) [reply README.md:87]: I don't understand why you are implying that we want to reject this PR.

Regarding the FAQ's path being incorrect, I used the exact same path you used in your previous commit (https://github.com/satisfactorymodding/ficsit-cli/commit/6d70a5ce606c29bf7030ba9295a04f5274ba6509#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R90). I have since updated it to the new path you made in another commit (https://github.com/satisfactorymodding/ficsit-cli/commit/2a45c6acbb59933ca82cdd9a1cdb624216b02ed5#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R93) after I made the original FAQ change. 

Users with the technical competence to set up custom install paths also have the patience to read a brief documentation passage and the understanding that others may not possess the same knowledge.

File: README.md [hunk 8]
```diff
@@ -78,6 +78,27 @@ To launch the interactive CLI, run the executable without any arguments.
 
 Run `ficsit help` to see a list of available commands.
 
+## Game Installations
```
Comment (Vilsol) [inline README.md:81]: This readme change is out of scope of this PR, as well as should be handled differently.

Rather, docs should have a subsection specifically on `ficsit-cli` just like it does for SMM: https://docs.ficsit.app/satisfactory-modding/latest/ForUsers/SatisfactoryModManager.html and the readme should reference that instead.

Please remove.

**Review by Vilsol — CHANGES_REQUESTED** [review]
Also please fix the linter issues

File: README.md (hunk 8 above)
Comment (kevinwright) [reply README.md:81]: Unless and until the docs have that relevant section, this README is the **only** place to discover that

- ficsit-cli, unlike SMM, does not auto-detect installations
- manually specifying the installation is a _necessary_ pre-requisite to using the app
- What the correct value for that will be in 99% of cases

Given that there's no way to add an installation via the interactive menus, it actually too me longer to figure out how to make the app work than it did to make the change.

This is the "boy scout" principle, leave a codebase in the state you would like to have found it, to make life easier for those that follow. I'm not at all comfortable reverting a change that achieves that, especially when the information it contains isn't something that's realistically ever going to change.

Comment (Vilsol) [reply README.md:81]: I'm not sure where you got the idea that it's not possible to add an installation via the interactive menu? I added a video below of how to do it.

As for the documentation, then please either go the extra length and include basic default instructions for all systems, or none. The most likely audience for this specific tool are Linux users running it on a headless server, where an SMM UI is impossible to be rendered.

I agree that the readme should be more informational, as it's currently even outdated for the paths it does provide, as it's missing the Linux equivalents, but also, this project is not in a released condition, but rather a 0-semver version, which should be considered unstable and not ready for general public use.

In general your message comes across quite passive-aggressive and condescending, which I am really not a fan of.

In either case, in its current state, I am not approving this PR.

(ignore weird colors for logo, I'm messing around with my terminal)

https://github.com/satisfactorymodding/ficsit-cli/assets/1759390/3fa176be-721b-4dc5-af9a-59b5531bcafd

Comment (kevinwright) [reply README.md:81]: Honestly, it's mostly frustration.

This PR is designed specifically for MacOS/Crossover users. I had to actively track down
- the fact that I had to manually add an installation
- the default installation path for satisfactory
- whether or not I needed to use the base `Satisfactory` folder or the nested `FactoryGame` folder

Earned knowledge that I chose to pay forward to future people who wanted to use this app. Requests for changes that essentially say "nope, don't tell others that, pass on the burden" don't make me feel at all comfortable.

**Comment by budak7273** [comment]
Superseded by #66
